service/src/main/java/bio/terra/externalcreds/services/NihCredentialsSyncService.java

```diff
@@ -0,0 +1,173 @@
+package bio.terra.externalcreds.services;
+
+import bio.terra.externalcreds.config.ExternalCredsConfig;
+import bio.terra.externalcreds.dataAccess.GoogleCloudStorageDAO;
+import bio.terra.externalcreds.exception.NihCredentialsSyncException;
+import bio.terra.externalcreds.models.NihCredentialsBlob;
+import bio.terra.externalcreds.models.NihCredentialsManifestEntry;
+import bio.terra.externalcreds.terra.FirecloudOrchestrationClient;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class NihCredentialsSyncService {
+
+  private final ExternalCredsConfig externalCredsConfig;
+  private final FirecloudOrchestrationClient firecloudOrchestrationClient;
+  private final GoogleCloudStorageDAO googleCloudStorageDAO;
+
+  public NihCredentialsSyncService(
+      ExternalCredsConfig externalCredsConfig,
+      FirecloudOrchestrationClient firecloudOrchestrationClient,
+      GoogleCloudStorageDAO googleCloudStorageDAO) {
+    this.externalCredsConfig = externalCredsConfig;
+    this.firecloudOrchestrationClient = firecloudOrchestrationClient;
+    this.googleCloudStorageDAO = googleCloudStorageDAO;
+  }
+
+  /**
+   * Check that all allowlists published by the NIH are still recent enough to be considered valid.
+   * The validity period is set in configs at
+   * `externalcreds.nih-credentials-sync-config.allowlist-validity-duration`
+   *
+   * <p>If `externalcreds.nih-credentials-sync-config.is-fail-closed` is set to true, then invalid
+   * allowlists will be overwritten with empty files, and Firecloud Orch will be told to update the
+   * Sam permissions, which will wipe out all access for the data governed by the allowlist.
+   *
+   * @return if all allowlists are valid or not
+   */
+  public boolean allAllowlistsValid() {
+    var invalidAllowlists = getInvalidAllowlists();
+    if (invalidAllowlists.isEmpty()) {
+      log.info("No NIH allowlists are outside of their validity period");
+      return true;
+    }
```

Comment:
More nits: Again, I would personally break this method down into multiple smaller methods.  If you would like examples, let me know.